crates/control/src/ball_state_composer.rs

```diff
@@ -41,7 +49,45 @@ impl BallStateComposer {
             context.team_ball,
             context.robot_to_field,
         ) {
-            (PrimaryState::Ready, ..) => None,
+            (PrimaryState::Ready, _, _, Some(robot_to_field)) => {
+                if let Some(game_controller_state) = context.game_controller_state {
+                    match game_controller_state {
+                        GameControllerState {
+                            sub_state: Some(SubState::PenaltyKick),
+                            kicking_team: Team::Opponent,
+                            ..
+                        } => Some(create_ball_state(
+                            robot_to_field.inverse()
+                                * point![
+                                    -context.field_dimensions.length / 2.0
+                                        + context.field_dimensions.penalty_marker_distance,
+                                    0.0
+                                ],
+                            context.robot_to_field,
+                            &mut self.last_ball_field_side,
+                            context.penalty_shot_direction.copied(),
+                        )),
+                        GameControllerState {
+                            sub_state: Some(SubState::PenaltyKick),
+                            kicking_team: Team::Hulks,
+                            ..
+                        } => Some(create_ball_state(
+                            robot_to_field.inverse()
+                                * point![
+                                    context.field_dimensions.length / 2.0
+                                        - context.field_dimensions.penalty_marker_distance,
+                                    0.0
+                                ],
+                            context.robot_to_field,
+                            &mut self.last_ball_field_side,
```

Comment:
I've changed it so only plus minus is different for the two but this is currently using another match statement across the entire game_controller_state. Please advise on how to clean up lines 63-66 so that I can just check the kicking team without having to involve the entire game_controller_state again

Reply:
Please see new, very much cleaner, code :)